support-models/src/main/scala/com/gu/support/catalog/ProductOptions.scala

```diff
@@ -1,44 +1,71 @@
 
-case object NoProductOptions extends ProductOptions
+sealed trait PaperProductOptions extends ProductOptions
 
-case object Saturday extends ProductOptions
+sealed trait DigitalProductOptions extends ProductOptions
 
-case object SaturdayPlus extends ProductOptions
+case object NoProductOptions extends ProductOptions with DigitalProductOptions
 
-case object Sunday extends ProductOptions
+case object Corporate extends DigitalProductOptions
 
-case object SundayPlus extends ProductOptions
+case object Saturday extends PaperProductOptions
 
-case object Weekend extends ProductOptions
+case object SaturdayPlus extends PaperProductOptions
 
-case object WeekendPlus extends ProductOptions
+case object Sunday extends PaperProductOptions
 
-case object SixdayPlus extends ProductOptions
+case object SundayPlus extends PaperProductOptions
 
-case object Sixday extends ProductOptions
+case object Weekend extends PaperProductOptions
 
-case object EverydayPlus extends ProductOptions
+case object WeekendPlus extends PaperProductOptions
 
-case object Everyday extends ProductOptions
+case object SixdayPlus extends PaperProductOptions
+
+case object Sixday extends PaperProductOptions
+
+case object EverydayPlus extends PaperProductOptions
+
+case object Everyday extends PaperProductOptions
 
 object ProductOptions {
-  val allProductOptions = List(Saturday, SaturdayPlus, Sunday, SundayPlus, Weekend, WeekendPlus, Sixday, SixdayPlus, Everyday, EverydayPlus)
+  val allProductOptions = DigitalProductOptions.productOptions ++ PaperProductOptions.productOptions
 
-  def fromString(code: String): Option[ProductOptions] = allProductOptions.find(_.getClass.getSimpleName == s"$code$$")
+  def fromString[T](code: String, productOptions: List[T]): Option[T] = productOptions.find(_.getClass.getSimpleName == s"$code$$")
 
   implicit val decode: Decoder[ProductOptions] =
-    Decoder.decodeString.emap(code => fromString(code).toRight(s"unrecognised product options '$code'"))
+    Decoder.decodeString.emap(code => fromString(code, allProductOptions).toRight(s"unrecognised product options '$code'"))
 
   implicit val encode: Encoder[ProductOptions] = Encoder.encodeString.contramap[ProductOptions](_.toString)
 
   implicit val keyEncoder: KeyEncoder[ProductOptions] = (productOptions: ProductOptions) => productOptions.toString
 
-  implicit val keyDecoder: KeyDecoder[ProductOptions] = (key: String) => fromString(key)
+  implicit val keyDecoder: KeyDecoder[ProductOptions] = (key: String) => fromString(key, allProductOptions)
+}
+
+object DigitalProductOptions {
+  val productOptions: List[catalog.DigitalProductOptions] = List(Corporate, NoProductOptions)
+
+  implicit val decoder: Decoder[DigitalProductOptions] =
+    Decoder.decodeString.emap(code => fromString(code, productOptions).toRight(s"unrecognised product options '$code'"))
+
+  implicit val encode: Encoder[DigitalProductOptions] = Encoder.encodeString.contramap[DigitalProductOptions](_.toString)
```

Comment:
using toString we would have to be careful about refactoring later - if we made into a case class for example, this would silently serialise to the wrong thing.  I know not all the serialisers are tested fully.
An alternative would be to give all the case objects a field for `serialisedValue` which I would usually do, but I can see why it could add boilerplate.

Reply:
This is the prevailing pattern in this code base although I've noticed that `ReaderType` does provide a serialisation value so I'm going to leave it like this for now. If we think there is sufficient value to changing this across the board then we should do that in a separate PR